crates/storage/provider/src/traits/stage_checkpoint.rs

```diff
@@ -0,0 +1,9 @@
+use reth_interfaces::Result;
+use reth_primitives::{stage::StageId, StageCheckpoint};
+
+/// The trait for fetching stage checkpoint related data.
+#[auto_impl::auto_impl(&, Arc)]
+pub trait StageCheckpointProvider: Send + Sync {
+    /// Fetch the checkpoint for the given stage.
+    fn get_stage_checkpoint(&self, id: StageId) -> Result<Option<StageCheckpoint>>;
```

Comment:
lgtm,

this is mainly useful for the Finish stage, so I think a default function that calls this with Finish would be useful